recipes/default.rb

```diff
@@ -22,6 +22,10 @@
 node.set[:exhibitor][:defaultconfig][:zoo_cfg_extra] = 'tickTime\=2000&initLimit\=60&syncLimit\=5'
 node.set[:exhibitor][:defaultconfig][:auto_manage_instances_settling_period_ms] = 1000
 
+if (node[:ipaddress].nil?)
+  node.set[:exhibitor][:opts][:hostname] = node[:network][:interfaces][node[:network][:default_interface]][:addresses].keys[1]
```

Comment:
@kgusakov you already referencing non-lo interface by `node[:network][:default_interface]` and trying to guess order of `addresses` for this default interface. I assume that you want IPv4 address, and not IPv6 or MAC. Given that you are sure that interface will have IPv4 address assigned you could get it via `select`:
```
chef > node[:network][:interfaces][node[:network][:default_interface]][:addresses].select{|k,v| v.family == "inet"}.keys[0]
 => "192.168.22.19"
chef >
```